oak_functions_client/src/lib.rs

```diff
@@ -14,26 +14,27 @@
 // limitations under the License.
 
 use anyhow::Context;
+use oak_attestation_verification::verifier::AttestationVerifier;
 use oak_client::{
-    proto::oak::session::v1::streaming_session_client::StreamingSessionClient,
-    transport::GrpcStreamingTransport, OakClient,
+    client::OakClient, proto::oak::session::v1::streaming_session_client::StreamingSessionClient,
+    transport::GrpcStreamingTransport,
 };
 use prost::Message;
 use tonic::transport::Channel;
 
 pub struct OakFunctionsClient {
-    oak_client: oak_client::OakClient<GrpcStreamingTransport>,
+    oak_client: OakClient<GrpcStreamingTransport>,
 }
 
 impl OakFunctionsClient {
-    pub async fn new(uri: &str) -> anyhow::Result<Self> {
+    pub async fn new<V: AttestationVerifier>(uri: &str, verifier: &V) -> anyhow::Result<Self> {
```

Comment:
I don't think we need a generic here, a Box dyn would probably be a bit simpler (also in OakClient::create), but up to you.

Reply:
I think so -- unless `verifier: & dyn AttestationVerifier` works?

Reply:
Done

Made it `verifier: &dyn AttestationVerifier`